src/bin/bots/watcher.cjs

```diff
@@ -0,0 +1,334 @@
+            "--notarize-errors",
+            "If set, eventual data retrieving errors will also get notarized."
+        )
+		.option(
+			"--signer <wit_pkh>",
+			"Witnet public key hash in charge of notarizing data updates.",
+			process.env.WITNET_SDK_WATCHER_WIT_SIGNER,
+		)
+        .option(
+			"--witnesses <number>",
+			"Size of the witnessing committee when notarizing data updates in Witnet.",
+			process.env.WITNET_SDK_WATCHER_WIT_WITNESSES,
+		)
+		
+	program.parse();
+
+    let { deviation } = program.opts();
+	const {
+		debug,
+        bytecode,
+        cooldown,
+        heartbeat,
+		minBalance,
+		network,
+        notarizeErrors,
+		priority,
+		signer,
+        witnesses
+	} = program.opts();
+
+	if (!debug) console.debug = () => {};
+
+	if (!WIT_WALLET_MASTER_KEY) {
+		console.error(
+			`❌ Fatal: a Witnet wallet's master key is not settled on  this environment.`,
+		);
+		process.exit(0);
+	}
+
+    const provider = network === "mainnet" ? "https://rpc-01.witnet.io" : (network === "testnet" ? "https://rpc-testnet.witnet.io" : network)
+	const wallet = await Witnet.Wallet.fromXprv(WIT_WALLET_MASTER_KEY, {
+		limit: 1,
+		provider: await Witnet.JsonRpcProvider.fromURL(provider),
+	});
+	const ledger = wallet.getSigner(signer || wallet.coinbase.pkh);
+	if (!ledger) {
+		console.error(
+			`❌ Fatal: hot wallet address ${signer} not found in wallet!`,
+		);
+		process.exit(0);
+	}
+
+    console.info(`Wit/RPC provider:  ${provider}`);
+	console.info(
+		`Witnet network:    WITNET:${wallet.provider.network.toUpperCase()} (${wallet.provider.networkId.toString(16)})`,
+	);
+	console.info(`Witnet hot wallet: ${ledger.pkh}`);
+	console.info(`Network priority:  ${priority.toUpperCase()}`);
+	console.info(
+		`Balance threshold: ${Witnet.Coins.fromWits(minBalance).toString(2)}`,
+	);
+    
+    if (!bytecode || !utils.isHexString(bytecode)) {
+        console.error(
+			`❌ Fatal: a valid Radon Request bytecode must be provided.`,
+		);
+		process.exit(0);
+    }
+
+    const request = Witnet.Radon.RadonRequest.fromBytecode(bytecode)
+    const dataType = request.dataType
+    const authorities = "(todo)"
+    console.info(`Radon bytecode:    ${bytecode}`)
+    console.info(`Radon RAD hash:    ${request.radHash}`)
+    console.info(`Radon data type:   ${dataType}`)
+    console.info(`Data authorities:  ${authorities}`)
+
+    // create DR transaction factory
+    const DRs = Witnet.DataRequests.from(ledger, request);
+
+    // validate deviation parameter, only on integer or float data feeds
+    if (["RadonInteger", "RadonFloat"].includes(request.dataType)) {
+        deviation = parseFloat(deviation || 0.0)
+        console.info(`Deviation thrshld: ${deviation.toFixed(2)} %`)
+    }
+    
+    // schedule signer's balance check
+	let balance = Witnet.Coins.fromPedros((await ledger.getBalance()).unlocked);
+	console.info(
+		`Initial balance:   ${balance.toString(2)}`,
+	);
+	if (balance.pedros < minBalance.pedros) {
```

Comment:
`balance.pedros` is a `BigInt` and `minBalance.pedros` comes from `Commander` and I'm not sure if it is using `BigInt` or `string` or `number`